test/test.js

```diff
@@ -292,11 +292,59 @@ describe('url-parse', function () {
     assume(parsed.pathname).equals('/foo/bar');
 
     url = 'https:/\\/\\/\\github.com/foo/bar';
+    parsed = parse(url);
     assume(parsed.host).equals('github.com');
     assume(parsed.hostname).equals('github.com');
     assume(parsed.pathname).equals('/foo/bar');
+
+    url = 'https:/github.com/foo/bar';
+    parsed = parse(url);
+    assume(parsed.host).equals('github.com');
+    assume(parsed.pathname).equals('/foo/bar');
+
+    url = 'https:\\github.com/foo/bar';
+    parsed = parse(url);
+    assume(parsed.host).equals('github.com');
+    assume(parsed.pathname).equals('/foo/bar');
+
+    url = 'https:github.com/foo/bar';
+    parsed = parse(url);
+    assume(parsed.host).equals('github.com');
+    assume(parsed.pathname).equals('/foo/bar');
+
+    url = 'https:github.com/foo/bar';
+    parsed = parse(url);
+    assume(parsed.host).equals('github.com');
+    assume(parsed.pathname).equals('/foo/bar');
   });
 
+  it('handles slashes after the protocol for non special URLs', function () {
+    var url = 'foo:example.com'
+      , parsed = parse(url);
+
+    assume(parsed.hostname).equals('');
+    assume(parsed.pathname).equals('example.com');
+    assume(parsed.href).equals('foo:example.com');
+
+    url = 'foo:/example.com';
+    parsed = parse(url);
+    assume(parsed.hostname).equals('');
+    assume(parsed.pathname).equals('/example.com');
+    assume(parsed.href).equals('foo:/example.com');
+
+    url = 'foo://example.com';
+    parsed = parse(url);
+    assume(parsed.hostname).equals('example.com');
+    assume(parsed.pathname).equals('/');
+    assume(parsed.href).equals('foo://example.com/');
```

Comment:
Behavior is not consistent with the Node.js WHATWG URL parser.

```
new URL('foo://example.com')
URL {
  href: 'foo://example.com',
  origin: 'null',
  protocol: 'foo:',
  username: '',
  password: '',
  host: 'example.com',
  hostname: 'example.com',
  port: '',
  pathname: '',
  search: '',
  searchParams: URLSearchParams {},
  hash: ''
}
```

```
> parse('foo://example.com')
{
  slashes: true,
  protocol: 'foo:',
  hash: '',
  query: '',
  pathname: '/',
  auth: '',
  host: 'example.com',
  port: '',
  hostname: 'example.com',
  password: '',
  username: '',
  origin: 'foo://example.com',
  href: 'foo://example.com/'
}
```

Our `pathname` is `'/'` instead of `''`.

Also note `origin` which should be `'null'` and not `'foo://example.com'`. It should be set only if the URL is special. Anyway this should be easy to fix in a different PR/commit.

Reply:
Fixed in https://github.com/unshiftio/url-parse/pull/208/commits/eda1342a352b980710e1947e7b0c617e53894310 and https://github.com/unshiftio/url-parse/pull/208/commits/e24fca31c61cc38f912c87225e28251d1b72599e.